src/gpu/intel/ocl/ocl_math_utils.h

```diff
@@ -690,9 +696,51 @@ float __attribute__((overloadable)) cvt_f4_e2m1_to_f32(uchar a) {
     return as_float((sign << 28) | (exp << 23) | (mant << 22));
 }
 
+#endif
+#if MATH_UTILS_DECLARE_F4_E3M0
+
+// OCL translation of common fp4 methods.
+uchar __attribute__((overloadable)) cvt_f32_to_f4_e3m0(float f) {
```

Comment:
I don't think it needs any action in this PR (reference implementation is complete), but for posterity: most supported data types have classes in `ocl_custom_types.h`, with the valid conversions/load/store signatures defined in `ocl_io.h`. Not all kernels use this interface yet, but eventually they will so they can get increased type safety.